### DISCUSSION TITLE
Plugin support

### DISCUSSION THREAD

**Discussion by zotroneneis** [astral-sh/rye#366] [Q&A]
Hi there,

I have general question about rye. 

I will be giving a talk on Python packaging at EuroPython next week ("An unbiased evaluation of environment management and packaging tools"). I would like to include rye in this talk. Since the talk evaluates all tools based on a predefined list of features, I would like to know if rye supports plugins.

I don't know much about Rust, so I'm not even sure if it would be possible to support plugins, or if the other applications would have to create rye plugins instead. Can someone help with this question?

If you would like to know more details about the talk you can find them [here](https://ep2023.europython.eu/session/an-unbiased-evaluation-of-environment-management-and-packaging-tools). A recording of an older version from PyConDE from April 2023 is [already on YouTube](https://www.youtube.com/watch?v=MsJjzVIVs6M). 

Thanks a lot for your help! :)

**Comment by bluss** [comment]
I know it's too late unfortunately, maybe I could try to answer. It's possible to create plugins in Rust but (to my knowledge) not a very developed art. I haven't seen anything in Rye issue tracker or docs that suggests a plugin system is planned, but it could be possible. No plugins right now. There's always different methods, for example that git supports just dropping in new binaries called git-something, that's like a limited plugin system.

I've written (as an experiment) plugins to tox and virtualenv, so the other way around, for them to support some rye features.
